### PR TITLE
Change logic for sending message to channel users

### PR DESCRIPTION
**Pull request by ThePhilgrim**
I think this improves the way messages are sent to channels. That way I don't have to repeat the same for loop in every single `handle_` function.

### CODE REVIEW

File: commands.py [hunk 1]
```diff
@@ -273,10 +272,8 @@ def handle_privmsg(state: mantatail.ServerState, user: mantatail.UserConnection,
     if user not in channel.users:
         error_not_on_channel(user, receiver)
     else:
-        for usr in channel.users:
-            if usr.nick != user.nick:
-                message = f"PRIVMSG {receiver} :{privmsg}"
-                usr.send_que.put((message, user.get_user_mask()))
+        privmsg_message = f"PRIVMSG {receiver} :{privmsg}"
+        channel.queue_message_to_chan_users(privmsg_message, user, False)
```
Comment (Akuli) [inline commands.py:276]: If you use a keyword argument, the reader doesn't have to guess what `False` does:

```suggestion
        channel.queue_message_to_chan_users(privmsg_message, user, send_to_self=False)
```

File: mantatail.py [hunk 2]
```diff
@@ -375,15 +375,23 @@ def is_founder(self, user: UserConnection) -> bool:
         """Checks if the user is the channel founder."""
         return user.user_name == self.founder
 
-    def kick_user(self, kicker: UserConnection, user_to_kick: UserConnection, message: str) -> None:
+    def kick_user(self, user_to_kick: UserConnection) -> None:
+        """Kicks user from the channel."""
+        self.users.discard(user_to_kick)
```
Comment (Akuli) [inline mantatail.py:380]: This method has now become so simple that it should just be removed IMO.

File: mantatail.py [hunk 3]
```diff
@@ -375,15 +375,23 @@ def is_founder(self, user: UserConnection) -> bool:
...
         """
         for usr in self.users:
-            usr.send_que.put((message, kicker.get_user_mask()))
-
-        self.users.discard(user_to_kick)
+            if not send_to_self:
+                if usr != sender:
+                    usr.send_que.put((message, sender.get_user_mask()))
+            else:
+                usr.send_que.put((message, sender.get_user_mask()))
```
Comment (Akuli) [inline mantatail.py:394]: Maybe it's simpler if you think about it like this: we want to send if it's a different user, or it's the same user but we are allowed to send to self.

```suggestion
            if usr != sender or send_to_self:
                usr.send_que.put((message, sender.get_user_mask()))
```

**Review by Akuli — COMMENTED** [review]
I like the idea! A few nits below.

File: commands.py (hunk 1 above)
Comment (ThePhilgrim) [reply commands.py:276]: Ah, nice! I didn't know that you could use the keyword in the argument too. I thought it was only in the parameters.

File: mantatail.py (hunk 3 above)
Comment (ThePhilgrim) [reply mantatail.py:394]: Very clever! I like it

**Pull request merged**